examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldServerS2A.java

```diff
@@ -0,0 +1,141 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldtls;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.grpc.Server;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyServerBuilder;
+import io.grpc.stub.StreamObserver;
+import io.netty.handler.ssl.ClientAuth;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.logging.Logger;
+
+// TODO: change flags to use com.google.common.flags
+
+/**
+ * Server that manages startup/shutdown of a {@code Greeter} server with TLS 1.3 enabled.
+ *
+ * args:
+ * [port]
+ * [certChainFilePath] File Path to Server Certificate
+ * [privateKeyFilePath] File Path to Server Private Key
+ * [trustCertCollectionFilePath] File Path to CA Certificate
+ */
+public class HelloWorldServerS2A {
+    private static final Logger logger = Logger.getLogger(HelloWorldServerS2A.class.getName());
+    private Server server;
+    private final int port;
+    private final String certChainFilePath;
+    private final String privateKeyFilePath;
+    private final String trustCertCollectionFilePath;
+
+    private HelloWorldServerS2A(int port,
+                             String certChainFilePath,
+                             String privateKeyFilePath,
+                             String trustCertCollectionFilePath) {
+        this.port = port;
+        this.certChainFilePath = certChainFilePath;
+        this.privateKeyFilePath = privateKeyFilePath;
+        this.trustCertCollectionFilePath = trustCertCollectionFilePath;
+    }
+
+    private SslContextBuilder getSslContextBuilder() {
+      SslContextBuilder sslClientContextBuilder = SslContextBuilder.forServer(new File(certChainFilePath), new File(privateKeyFilePath));
+        if (trustCertCollectionFilePath != null) {
+            sslClientContextBuilder.trustManager(new File(trustCertCollectionFilePath));
+            sslClientContextBuilder.clientAuth(ClientAuth.REQUIRE);
+            sslClientContextBuilder.protocols("TLSv1.3");
+        }
+        return GrpcSslContexts.configure(sslClientContextBuilder);
+    }
+
+    private void start() throws IOException {
+        server = NettyServerBuilder.forPort(port)
+                .sslContext(getSslContextBuilder().build())
+                .addService(new GreeterImpl())
+                .build()
+                .start();
+
+        logger.info("Server started, listening on " + port);
```

Comment:
s/listening at address/host:port

Reply:
Please include the host in the address that is logged.

Reply:
the server host is defined in its cert, right? Is it possible to extract the info? 

If we were to instead include the host as an argument that is passed into main, then it would be inconsistent from that of Go, who's flags also only consist of the port...

Thoughts?

Reply:
For now, you can just hardcode the host here as `localhost`.